CMakeLists.txt

```diff
@@ -186,6 +188,36 @@ add_library(PKWare STATIC
 add_library(SourceS STATIC
   SourceS/paths.cpp)
 
+if(DISCORD_INTEGRATION AND WIN32)

```

Comment:
Let's test in a separate `if` that the platform is supported and error out explicitly if it isn't. Then we can remove `AND WIN32` throughout